tests/ui/panicking_macros.rs

```diff
@@ -1,8 +1,23 @@
-#![warn(clippy::unimplemented, clippy::unreachable, clippy::todo, clippy::panic)]
 #![allow(clippy::assertions_on_constants, clippy::eq_op)]
+#![feature(inline_const)]
+#![warn(clippy::unimplemented, clippy::unreachable, clippy::todo, clippy::panic)]
 
 extern crate core;
 
+const _: () = {
+    if 1 == 0 {
+        panic!("A balanced diet means a cupcake in each hand");
+    }
+};
+
+fn inline_const() {
+    let _ = const {
```

Comment:
Why not just make the function `const`? No need to enable a feature in a test.

Reply:
Constant functions can be called in non-constant environments, therefore, they can still panic at run-time. On the other hand, inline blocks will always be evaluated at compile-time.

Reply:
Ah right, now I also understand what this PR is about. :sweat_smile: